File: lib/array/arrGroupByMaxCount.ts
export function arrGroupByMaxCount(array: any[], maxLengthPerGroup = 1): any[] {
	const result = [];
	const arrayLength = array.length;
	let tempArray = [];

	for (let i = 0; i < arrayLength; i += 1) {
		if (tempArray.length === maxLengthPerGroup) {
			result.push(tempArray);
			tempArray = [];
		}

		tempArray.push(array[i]);
	}

	if (tempArray.length > 0) {
		result.push(tempArray);
	}

	return result;
}

File: packages/python/qsu/array/arrGroupByMaxCount.py
def arrGroupByMaxCount(array: list, maxLengthPerGroup: int = 1) -> list:
	result = []
	tempArray = []

	for item in array:
		if len(tempArray) == maxLengthPerGroup:
			result.append(tempArray)
			tempArray = []

		tempArray.append(item)

	if len(tempArray) > 0:
		result.append(tempArray)

	return result
